packages/avocode-email-tagsinput/test/e2e/test-utils.js

```diff
@@ -0,0 +1,61 @@
+const tagsInputFieldSelector = '.tagsinput'
+const tagSubmitKeys = [ 'Comma', 'Enter', 'Space' ]
+const URLRoot = isDevelopment
+  ? 'http://localhost:8080/#/avocode-email-tagsinput'
+  : 'https://avocode-email-tagsinput.surge.sh/#/avocode-email-tagsinput'
+
+const getInputText = async (page, inputFieldSelector = tagsInputFieldSelector) => {
+  const text = await page.evaluate((selector) => {
+    const editor = document.querySelector(selector)
+
+    return editor.textContent
+  }, inputFieldSelector)
+
+  return text.trim()
+}
+
+const getTagNodes = async (page, inputFieldSelector = tagsInputFieldSelector) => {
+  return page.evaluate((selector) => {
+    const editor = document.querySelector(selector)
+    const tagNodes = editor.querySelectorAll('.tag')
+
+    return Array.from(tagNodes)
+      .map((node) => {
+        return {
+          text: node.textContent.trim(),
+          className: node.className,
+          html: node.innerHTML,
+        }
+      })
+  }, inputFieldSelector)
+}
+
+
+const getAnchorAndFocus = async (page) => {
+  return page.evaluate(() => {
+    const selection = window.getSelection()
+
+    return [ selection.anchorOffset, selection.focusOffset ]
+  })
+}
+
+const addEmailTag = async (page, emailTag, submitKey = 'Comma') => {
+  await page.click(tagsInputFieldSelector)
+  await page.type(tagsInputFieldSelector, emailTag)
+  if (tagSubmitKeys.includes(submitKey)) {
```

Comment:
Not sure why you would do this kind of check here. It's not a concern of testing utility to check for "valid" keys because this library does not have concept of valid keys. You can use `addTagKeyCodes` to pass any key that will be used for adding tags, yes even nonsensical but I saw no point in enforcing it.

Comma, enter and space are just meaningful defaults that can be overriden. Also there's Flow type checker that will check for numeric type so no point in this really.